src/app/client/src/app/modules/dashboard/components/course-progress/course-progress.component.ts

```diff
@@ -449,21 +547,70 @@ export class CourseProgressComponent implements OnInit, OnDestroy, AfterViewInit
     };
     this.courseProgressService.getReportsMetaData(requestParams).subscribe((response) => {
       if (_.get(response, 'responseCode') === 'OK') {
-        this.progressReportUpdatedOn =  _.get(response, 'result.course-progress-reports.lastModified') || null;
+        this.progressReportUpdatedOn = _.get(response, 'result.course-progress-reports.lastModified') || null;
         this.scoreReportUpdatedOn = _.get(response, 'result.assessment-reports.lastModified') || null;
       }
     });
   }
+
+
+
+  getFieldValue(array, field) {
+    return _.find(array, { "type": field }).count;
+  }
+
   /**
   * To method subscribes the user data to get the user id.
   * It also subscribes the activated route params to get the
   * course id and timeperiod
   */
   ngOnInit() {
+    // ---- Mock data Start-----
+    const apiData = {
+      userConsent: 'No',
+      audience: 'Teacher'
+    };
+
+    this.user.userData$.subscribe((user) => {
+      const userProfile = user.userProfile;
+      let userRoles = _.get(userProfile, 'userRoles');
+      // userRoles = ['COURSE_MENTOR']
+      const isCourseCreator = _.includes(userRoles, 'COURSE_CREATOR');
+      const formReadInputParams = {
+        formType: 'batch',
+        formAction: 'list',
+        contentType: 'report_types'
+      };
+      this.formService.getFormConfig(formReadInputParams).subscribe(
```

Comment:
Why are me calling this on ngOnit? Call only on the selection of "detailed report" tab or on batch change when the user in "detailed report" tab.